### PR TITLE
Android build document in English

### PR DESCRIPTION
**Pull request by wangkuiyi**

### CODE REVIEW

File: doc/howto/cross_compiling/cross_compiling_for_android.md [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+          --arch=arm64 --platform=android-21 --install-dir=your/path/to/arm64_standalone_toolchain
+  ```
+
+The generated toolchain will be in `your/path/to/arm64_standalone_toolchain`.
+
+**Please be aware that the minimum level of Android API required by PaddlePaddle is 21.**
+
+### Cross-Compiling Arguments
+
+CMake supports to [choose the toolchain](https://cmake.org/cmake/help/v3.0/manual/cmake-toolchains.7.html#cross-compiling).  PaddlePaddle provides [`android.cmake`](https://github.com/PaddlePaddle/Paddle/blob/develop/cmake/cross_compiling/android.cmake), which configures the Android cross-compiling toolchain for CMake.  `android.cmake` is not required for CMake >= 3.7, which starts supporting Android cross-compiling.  。PaddlePaddle detects CMake version, for those newer than 3.7, it uses [the official version](https://cmake.org/cmake/help/v3.7/manual/cmake-toolchains.7.html#cross-compiling).
```
Comment (Xreki) [inline doc/howto/cross_compiling/cross_compiling_for_android.md]: I am a little confused about the meaning of `choose the toolchain`. I'd like to change it to `CMake supports to cross-compile`.
There is a extra `。`

File: doc/howto/cross_compiling/cross_compiling_for_android.md [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+
+Some other CMake parameters you might want to know:
```
Comment (Xreki) [inline doc/howto/cross_compiling/cross_compiling_for_android.md]: parameters -> arguments?
Some other CMake parameters you might want to know -> Some CMake arguments you have to know

File: doc/howto/cross_compiling/cross_compiling_for_android.md [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+
+- `CMAKE_SYSTEM_NAME` must be `Android`.  This tells PaddlePaddle's CMake system to cross-compile dependencies including protobuf library and the protoc executable, and ARM-enabled OpenBLAS.  This also changes some other CMake parameters like `WITH_GPU=OFF`, `WITH_AVX=OFF`, `WITH_PYTHON=OFF`, and `WITH_RDMA=OFF`.
```
Comment (Xreki) [inline doc/howto/cross_compiling/cross_compiling_for_android.md]: `protobuf library and the protoc executable, and ARM-enabled OpenBLAS`, maybe detail is not needed here. Most people don't need to know the compiling process.

File: doc/howto/cross_compiling/cross_compiling_for_android.md [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+- `CMAKE_BUILD_TYPE-Release` optimizes the runtime performance.
+
+Our own tip for performance optimization to use clang and Eigen or OpenBLAS:
+- `CMAKE_BUILD_TYPE=Release`
+- `ANDROID_TOOLCHAIN=clang`
+- `USE_EIGEN_BLAS=ON` for `armeabi-v7a`, or `USE_EIGEN_FOR_BLAS=OFF` for `arm64-v8a`.
+
+### Build and Install
+
+After running `cmake`, we can run `make install` to build and install.
```
Comment (Xreki) [inline doc/howto/cross_compiling/cross_compiling_for_android.md]: `make install` -> `make; make install`

File: doc/howto/cross_compiling/cross_compiling_for_android.md [hunk 5]
```diff
@@ -0,0 +1,153 @@
...
+
+Before building, you might want to remove the `third_party` and `build` directories including pre-built libraries for other ARM architectures.
```
Comment (Xreki) [inline doc/howto/cross_compiling/cross_compiling_for_android.md]: for other ARM architectures -> for other architectures

File: doc/howto/cross_compiling/cross_compiling_for_android.md [hunk 6]
```diff
@@ -0,0 +1,153 @@
...
+1. Use a Docker container, or
+2. cross-compiling on Linux.
+
+## Docker-based Cross-Compiling
+
+Docker-based cross-compiling is the recommended approach because Docker runs on all major operating systems, including Linux, Mac OS X, and Windows.
+
+### Build the Docker Image
+
+The following steps pack tools we need to build PaddlePaddle into a Docker image.
```
Comment (kavyasrinet) [inline doc/howto/cross_compiling/cross_compiling_for_android.md]: Rephrase: The following steps pack all the tools that we need to build PaddlePaddle, into a Docker image.

File: doc/howto/cross_compiling/cross_compiling_for_android.md [hunk 7]
```diff
@@ -0,0 +1,153 @@
...
+
+```bash
+$ git clone https://github.com/PaddlePaddle/Paddle.git
+$ cd Paddle
+$ docker build -t paddle:dev-android . -f Dockerfile.android
+```
+
+### Build the Inference Library
+
+We can run this Docker image to build PaddlePaddle's inference library for Android:
```
Comment (kavyasrinet) [inline doc/howto/cross_compiling/cross_compiling_for_android.md]: Rephrase: We can run the Docker image we just created to build the inference library for PaddlePaddle for Android using the command below:

File: doc/howto/cross_compiling/cross_compiling_for_android.md [hunk 8]
```diff
@@ -0,0 +1,153 @@
...
+The Docker image accepts two arguments `ANDROID_ABI` and `ANDROID_API`:
+
+| Argument        | Optional Values         | Default |
+|-----------------|-------------------------|---------|
+|`ANDROID_ABI`    |`armeabi-v7a, arm64-v8a` | `armeabi-v7a` |
+|`ANDROID_API`    |`>= 21` | `21` |
+
+The ARM-64 architecture (`arm64-v8a`) requires at least level 21 of Android API.
+
+The default entry-point of the Docker image,[`paddle/scripts/docker/build_android.sh`](https://github.com/PaddlePaddle/Paddle/blob/develop/paddle/scripts/docker/build_android.sh) generates the [Android cross-compiling standalone toolchain](https://developer.android.com/ndk/guides/standalone_toolchain.html) according to the value of `ANDROID_ABI` and `ANDROID_API`.  For information about other configuration parameters, please go on reading.
```
Comment (kavyasrinet) [inline doc/howto/cross_compiling/cross_compiling_for_android.md]: Rephrase : The default entry-point of the Docker image [`paddle/scripts/docker/build_android.sh`](https://github.com/PaddlePaddle/Paddle/blob/develop/paddle/scripts/docker/build_android.sh) generates the [Android cross-compiling standalone toolchain](https://developer.android.com/ndk/guides/standalone_toolchain.html) based on the argument : `ANDROID_ABI` or `ANDROID_API`.  For more information about other configuration parameters, please continue reading.

File: doc/howto/cross_compiling/cross_compiling_for_android.md [hunk 9]
```diff
@@ -0,0 +1,153 @@
...
+
+Above command line generates and outputs the inference library in `$PWD/install_android` and puts third-party libraries in `$PWD/install_android/third_party`.
```
Comment (kavyasrinet) [inline doc/howto/cross_compiling/cross_compiling_for_android.md]: Rephrase: The above command generates and outputs the inference library in `$PWD/install_android` and puts third-party libraries in `$PWD/install_android/third_party`.

File: doc/howto/cross_compiling/cross_compiling_for_android.md [hunk 10]
```diff
@@ -0,0 +1,153 @@
...
+
+## Linux-based Cross-Compiling
+
+The Linux-base approach is to run steps in `Dockerfile.android` manually on a Linux x64 computer.
```
Comment (kavyasrinet) [inline doc/howto/cross_compiling/cross_compiling_for_android.md]: Rephrase: The Linux-based approach to cross-compile is to run steps in `Dockerfile.android` manually on a Linux x64 computer.

File: doc/howto/cross_compiling/cross_compiling_for_android.md (hunk 1 above)
Comment (kavyasrinet) [inline doc/howto/cross_compiling/cross_compiling_for_android.md]: Rephrase: CMake supports [choosing the toolchain](https://cmake.org/cmake/help/v3.0/manual/cmake-toolchains.7.html#cross-compiling).  PaddlePaddle provides [`android.cmake`](https://github.com/PaddlePaddle/Paddle/blob/develop/cmake/cross_compiling/android.cmake), which configures the Android cross-compiling toolchain for CMake.  `android.cmake` is not required for CMake versions >= 3.7, which support Android cross-compiling.  PaddlePaddle detects the CMake version, for those newer than 3.7, it uses [the official version](https://cmake.org/cmake/help/v3.7/manual/cmake-toolchains.7.html#cross-compiling).

File: doc/howto/cross_compiling/cross_compiling_for_android.md [hunk 11]
```diff
@@ -0,0 +1,153 @@
...
+
+Some other CMake parameters you might want to know:
+
+- `CMAKE_SYSTEM_NAME` must be `Android`.  This tells PaddlePaddle's CMake system to cross-compile dependencies including protobuf library and the protoc executable, and ARM-enabled OpenBLAS.  This also changes some other CMake parameters like `WITH_GPU=OFF`, `WITH_AVX=OFF`, `WITH_PYTHON=OFF`, and `WITH_RDMA=OFF`.
+- `WITH_C_API` must be `ON`, to build the C-based inference library for Android.
+- `WITH_SWIG_PY` must be `OFF` because the Android platform doesn't support SWIG-based API.
+
+Some Android-specific parameters:
+
+- `ANDROID_STANDALONE_TOOLCHAIN` the absolute path of the Android standalone toolchain, or the path relative to the CMake build directory.  PaddlePaddle's CMake extensions would derive the cross-compiler, sysroot and Android API level from this parameter.
```
Comment (kavyasrinet) [inline doc/howto/cross_compiling/cross_compiling_for_android.md]: `ANDROID_STANDALONE_TOOLCHAIN` => `ANDROID_STANDALONE_TOOLCHAIN`:

File: doc/howto/cross_compiling/cross_compiling_for_android.md [hunk 12]
```diff
@@ -0,0 +1,153 @@
...
+- `ANROID_ARM_MODE`
+  - could be `ON` or `OFF`, and defaults to `ON`, when `ANDROID_ABI=armeabi-v7a`;
+  - no need to specify when `ANDROID_ABI=arm64-v8a`.
+- `ANDROID_ARM_NEON` indicates if to use NEON instructions.
+  - could be `ON` or `OFF`, and defaults to `ON`, when `ANDROID_ABI=armeabi-v7a`;
+  - no need to specify when `ANDROID_ABI=arm64-v8a`.
+
+Other useful parameters:
+
+- `USE_EIGEN_FOR_BLAS` indicates if using Eigen.  Could be `ON` or `OFF`, defaults to `OFF`.
```
Comment (kavyasrinet) [inline doc/howto/cross_compiling/cross_compiling_for_android.md]: `USE_EIGEN_FOR_BLAS` => `USE_EIGEN_FOR_BLAS`:

File: doc/howto/cross_compiling/cross_compiling_for_android.md [hunk 13]
```diff
@@ -0,0 +1,153 @@
...
+- `HOST_C/CXX_COMPILER` specifies the host compiler, which is used to build the host-specific protoc and target-specific OpenBLAS.  It defaults to the value of the environment variable `CC`, or `cc`.
```
Comment (kavyasrinet) [inline doc/howto/cross_compiling/cross_compiling_for_android.md]: `HOST_C/CXX_COMPILER` => `HOST_C/CXX_COMPILER`:

File: doc/howto/cross_compiling/cross_compiling_for_android.md [hunk 14]
```diff
@@ -0,0 +1,153 @@
+# Build PaddlePaddle for Android
+
+There are two approaches to build PaddlePaddle for Android:
+
+1. Use a Docker container, or
+2. cross-compiling on Linux.
```
[the remark this answers is not in the document]
Comment (wangkuiyi) [reply doc/howto/cross_compiling/cross_compiling_for_android.md]: Got it. Done

File: doc/howto/cross_compiling/cross_compiling_for_android.md (hunk 8 above)
[the remark this answers is not in the document]
Comment (wangkuiyi) [reply doc/howto/cross_compiling/cross_compiling_for_android.md]: Done

File: doc/howto/cross_compiling/cross_compiling_for_android.md [hunk 15]
```diff
@@ -0,0 +1,153 @@
...
+
+Above command line generates and outputs the inference library in `$PWD/install_android` and puts third-party libraries in `$PWD/install_android/third_party`.
+
+## Linux-based Cross-Compiling
+
+The Linux-base approach is to run steps in `Dockerfile.android` manually on a Linux x64 computer.
+
+### Setup the Environment
+
+To install Android's C/C++ cross-compiling standalone toolchain, we need [Android NDK](https://developer.android.com/ndk/downloads/index.html?hl=zh-cn):
```
[the remark this answers is not in the document]
Comment (wangkuiyi) [reply doc/howto/cross_compiling/cross_compiling_for_android.md]: Done

File: doc/howto/cross_compiling/cross_compiling_for_android.md [hunk 16]
```diff
@@ -0,0 +1,153 @@
...
+
+```bash
+wget -q https://dl.google.com/android/repository/android-ndk-r14b-linux-x86_64.zip
+unzip -q android-ndk-r14b-linux-x86_64.zip
+```
+
+Android NDK includes everything we need to build the cross-compiling toolchain.
```
[the remark this answers is not in the document]
Comment (wangkuiyi) [reply doc/howto/cross_compiling/cross_compiling_for_android.md]: OK

File: doc/howto/cross_compiling/cross_compiling_for_android.md [hunk 17]
```diff
@@ -0,0 +1,153 @@
...
+
+- To build the toolchain for `armeabi-v7a` and Android API level 21:
```
[the remark this answers is not in the document]
Comment (wangkuiyi) [reply doc/howto/cross_compiling/cross_compiling_for_android.md]: Got it. Done.

File: doc/howto/cross_compiling/cross_compiling_for_android.md [hunk 18]
```diff
@@ -0,0 +1,153 @@
...
+
+  ```bash
+  your/path/to/android-ndk-r14b-linux-x86_64/build/tools/make-standalone-toolchain.sh \
+          --arch=arm --platform=android-21 --install-dir=your/path/to/arm_standalone_toolchain
+  ```
+
+- To build the toolchain for `arm64-v8a` and Android API level 21:
```
[the remark this answers is not in the document]
Comment (wangkuiyi) [reply doc/howto/cross_compiling/cross_compiling_for_android.md]: Done

File: doc/howto/cross_compiling/cross_compiling_for_android.md (hunk 1 above)
Comment (wangkuiyi) [reply doc/howto/cross_compiling/cross_compiling_for_android.md]: Done.

File: doc/howto/cross_compiling/cross_compiling_for_android.md (hunk 2 above)
Comment (wangkuiyi) [reply doc/howto/cross_compiling/cross_compiling_for_android.md]: Done

File: doc/howto/cross_compiling/cross_compiling_for_android.md (hunk 3 above)
Comment (wangkuiyi) [reply doc/howto/cross_compiling/cross_compiling_for_android.md]: Done

File: doc/howto/cross_compiling/cross_compiling_for_android.md (hunk 4 above)
Comment (wangkuiyi) [reply doc/howto/cross_compiling/cross_compiling_for_android.md]: Done.  But usually, projects can be built using a single command `make install`.

File: doc/howto/cross_compiling/cross_compiling_for_android.md (hunk 5 above)
Comment (wangkuiyi) [reply doc/howto/cross_compiling/cross_compiling_for_android.md]: Done

File: doc/howto/cross_compiling/cross_compiling_for_android.md (hunk 6 above)
Comment (wangkuiyi) [reply doc/howto/cross_compiling/cross_compiling_for_android.md]: Done

File: doc/howto/cross_compiling/cross_compiling_for_android.md (hunk 7 above)
Comment (wangkuiyi) [reply doc/howto/cross_compiling/cross_compiling_for_android.md]: Done

File: doc/howto/cross_compiling/cross_compiling_for_android.md (hunk 8 above)
Comment (wangkuiyi) [reply doc/howto/cross_compiling/cross_compiling_for_android.md]: Done

File: doc/howto/cross_compiling/cross_compiling_for_android.md (hunk 9 above)
Comment (wangkuiyi) [reply doc/howto/cross_compiling/cross_compiling_for_android.md]: Done

File: doc/howto/cross_compiling/cross_compiling_for_android.md (hunk 10 above)
Comment (wangkuiyi) [reply doc/howto/cross_compiling/cross_compiling_for_android.md]: Done

File: doc/howto/cross_compiling/cross_compiling_for_android.md (hunk 1 above)
Comment (wangkuiyi) [reply doc/howto/cross_compiling/cross_compiling_for_android.md]: Done

File: doc/howto/cross_compiling/cross_compiling_for_android.md (hunk 11 above)
Comment (wangkuiyi) [reply doc/howto/cross_compiling/cross_compiling_for_android.md]: Done

File: doc/howto/cross_compiling/cross_compiling_for_android.md (hunk 12 above)
Comment (wangkuiyi) [reply doc/howto/cross_compiling/cross_compiling_for_android.md]: Done

File: doc/howto/cross_compiling/cross_compiling_for_android.md (hunk 13 above)
Comment (wangkuiyi) [reply doc/howto/cross_compiling/cross_compiling_for_android.md]: Done

**Comment by Xreki** [comment]
I merge this PR and will give a PR to rename to file.

**Pull request merged**